ui/app/models/kv/metadata.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Model, { attr } from '@ember-data/model';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+import { keyIsFolder } from 'core/utils/key-utils';
+
+const validations = {
+  maxVersions: [
+    { type: 'number', message: 'Maximum versions must be a number.' },
+    { type: 'length', options: { min: 1, max: 16 }, message: 'You cannot go over 16 characters.' },
+  ],
+};
+const formFieldProps = ['customMetadata', 'maxVersions', 'casRequired', 'deleteVersionAfter'];
+
+@withModelValidations(validations)
+@withFormFields(formFieldProps)
+export default class KvSecretMetadataModel extends Model {
+  @attr('string') backend;
+  @attr('string') path;
+  @attr('string') fullSecretPath;
+
+  @attr('number', {
+    defaultValue: 0,
```

Comment:
Didn't we fix an issue with this recently, do we need to remove this value?

Reply:
I _think_ you're correct, we can remove the default here. @hellobontempo is that correct? [Here's the PR](https://github.com/hashicorp/vault/pull/22394/files) I think we're talking about.